File: backend/app/routers/inspections.py
from fastapi import APIRouter, Depends, HTTPException, status, UploadFile, File
from sqlalchemy.orm import Session
from typing import List, Optional
from datetime import datetime, date
from pydantic import BaseModel

from ..database import get_db
from ..auth import get_current_active_user, require_roles
from ..models.user import User
from ..models.unit import Unit
from ..models.property import Property
from ..models.enums import UnitStatus

router = APIRouter(prefix="/inspections", tags=["Inspections"])

# Pydantic models for inspection
class InspectionBase(BaseModel):
    unit_id: int
    inspection_date: date
    inspection_type: str  # routine, move_in, move_out, emergency
    inspector_name: str
    status: str  # scheduled, in_progress, completed, cancelled
    notes: Optional[str] = None
    findings: Optional[str] = None
    recommendations: Optional[str] = None

class InspectionCreate(InspectionBase):
    pass

class InspectionUpdate(BaseModel):
    inspection_date: Optional[date] = None
    inspection_type: Optional[str] = None
    inspector_name: Optional[str] = None
    status: Optional[str] = None
    notes: Optional[str] = None
    findings: Optional[str] = None
    recommendations: Optional[str] = None

class InspectionResponse(InspectionBase):
    id: int
    unit_number: str
    property_name: str
    property_address: str
    monthly_rent: float
    bedrooms: int
    bathrooms: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# Mock data storage (in real app, this would be a database table)
inspections_db = [
    {
        "id": 1,
        "unit_id": 1,
        "unit_number": "101",
        "property_name": "Sunset Apartments",
        "property_address": "123 Sunset Boulevard, Los Angeles, CA",
        "inspection_date": "2025-10-25",
        "inspection_type": "routine",
        "inspector_name": "John Smith",
        "status": "scheduled",
        "notes": "Quarterly routine inspection",
        "findings": "",
        "recommendations": "",
        "monthly_rent": 2500.0,
        "bedrooms": 2,
        "bathrooms": 2,
        "created_at": "2025-10-20T10:00:00Z",
        "updated_at": "2025-10-20T10:00:00Z"
    },
    {
        "id": 2,
        "unit_id": 2,
        "unit_number": "102",
        "property_name": "Sunset Apartments",
        "property_address": "123 Sunset Boulevard, Los Angeles, CA",
        "inspection_date": "2025-10-28",
        "inspection_type": "move_out",
        "inspector_name": "Jane Doe",
        "status": "completed",
        "notes": "Move-out inspection for departing tenant",
        "findings": "Minor wall damage, carpet needs cleaning",
        "recommendations": "Repair wall damage, deep clean carpet",
        "monthly_rent": 2200.0,
        "bedrooms": 1,
        "bathrooms": 1,
        "created_at": "2025-10-18T14:30:00Z",
        "updated_at": "2025-10-20T15:45:00Z"
    }
]

@router.get("/", response_model=List[InspectionResponse])
async def get_inspections(
    status: Optional[str] = None,
    inspection_type: Optional[str] = None,
    current_user: User = Depends(require_roles(["admin", "owner", "manager"])),
    db: Session = Depends(get_db)
):
    """Get all inspections with optional filtering"""
    try:
        # Filter inspections based on user role
        if current_user.role == "owner":
            # Get owner's properties
            owner_properties = db.query(Property).filter(Property.owner_id == current_user.id).all()
            property_ids = [p.id for p in owner_properties]
            
            # Filter inspections for owner's properties only
            filtered_inspections = [
                inspection for inspection in inspections_db
                if any(unit.property_id in property_ids for unit in db.query(Unit).filter(Unit.id == inspection["unit_id"]))
            ]
        else:
            filtered_inspections = inspections_db
        
        # Apply additional filters
        if status:
            filtered_inspections = [i for i in filtered_inspections if i["status"] == status]
        
        if inspection_type:
            filtered_inspections = [i for i in filtered_inspections if i["inspection_type"] == inspection_type]
        
        return filtered_inspections
        
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error fetching inspections: {str(e)}"
        )

@router.get("/available-units", response_model=List[dict])
async def get_available_units_for_inspection(
    current_user: User = Depends(require_roles(["admin", "owner", "manager"])),
    db: Session = Depends(get_db)
):
    """Get all units available for rent that can be inspected"""
    try:
        if current_user.role == "owner":
            # Get owner's properties
            owner_properties = db.query(Property).filter(Property.owner_id == current_user.id).all()
            property_ids = [p.id for p in owner_properties]
            
            # Get available units from owner's properties
            available_units = db.query(Unit).filter(
                Unit.property_id.in_(property_ids),
                Unit.status == UnitStatus.AVAILABLE
            ).all()
        else:
            # Admin/Manager can see all available units
            available_units = db.query(Unit).filter(Unit.status == UnitStatus.AVAILABLE).all()
        
        # Format response
        units_data = []
        for unit in available_units:
            property_obj = db.query(Property).filter(Property.id == unit.property_id).first()
            units_data.append({
                "id": unit.id,
                "unit_number": unit.unit_number,
                "property_id": unit.property_id,
                "property_name": property_obj.name if property_obj else "Unknown",
                "property_address": f"{property_obj.address}, {property_obj.city}" if property_obj else "Unknown",
                "monthly_rent": float(unit.monthly_rent),
                "bedrooms": unit.bedrooms,
                "bathrooms": unit.bathrooms,
                "unit_type": unit.unit_type.value,
                "floor": unit.floor,
                "amenities": unit.amenities,
                "description": unit.description
            })
        
        return units_data
        
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error fetching available units: {str(e)}"
        )

@router.post("/", response_model=InspectionResponse)
async def create_inspection(
    inspection: InspectionCreate,
    current_user: User = Depends(require_roles(["admin", "owner"])),
    db: Session = Depends(get_db)
):
    """Create a new inspection for a rental unit"""
    try:
        # Verify unit exists and user has access
        unit = db.query(Unit).filter(Unit.id == inspection.unit_id).first()
        if not unit:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail="Unit not found"
            )
        
        # CRITICAL: Only allow inspections for units that are AVAILABLE for rent
        if unit.status != UnitStatus.AVAILABLE:
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail=f"Inspections can only be scheduled for units that are available for rent. This unit is currently {unit.status.value}."
            )
        
        # Check if user has access to this unit
        if current_user.role == "owner":
            property_obj = db.query(Property).filter(Property.id == unit.property_id).first()
            if not property_obj or property_obj.owner_id != current_user.id:
                raise HTTPException(
                    status_code=status.HTTP_403_FORBIDDEN,
                    detail="Access denied"
                )
        
        # Get property information
        property_obj = db.query(Property).filter(Property.id == unit.property_id).first()
        
        # Create new inspection
        new_inspection = {
            "id": len(inspections_db) + 1,
            "unit_id": inspection.unit_id,
            "unit_number": unit.unit_number,
            "property_name": property_obj.name if property_obj else "Unknown",
            "property_address": f"{property_obj.address}, {property_obj.city}" if property_obj else "Unknown",
            "inspection_date": inspection.inspection_date.isoformat(),
            "inspection_type": inspection.inspection_type,
            "inspector_name": inspection.inspector_name,
            "status": inspection.status,
            "notes": inspection.notes,
            "findings": inspection.findings,
            "recommendations": inspection.recommendations,
            "monthly_rent": float(unit.monthly_rent),
            "bedrooms": unit.bedrooms,
            "bathrooms": unit.bathrooms,
            "created_at": datetime.utcnow().isoformat(),
            "updated_at": datetime.utcnow().isoformat()
        }
        
        inspections_db.append(new_inspection)
        
        return new_inspection
        
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error creating inspection: {str(e)}"
        )

@router.get("/{inspection_id}", response_model=InspectionResponse)
async def get_inspection(
    inspection_id: int,
    current_user: User = Depends(require_roles(["admin", "owner", "manager"])),
    db: Session = Depends(get_db)
):
    """Get a specific inspection by ID"""
    try:
        inspection = next((i for i in inspections_db if i["id"] == inspection_id), None)
        if not inspection:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail="Inspection not found"
            )
        
        # Check access permissions
        if current_user.role == "owner":
            unit = db.query(Unit).filter(Unit.id == inspection["unit_id"]).first()
            if unit:
                property_obj = db.query(Property).filter(Property.id == unit.property_id).first()
                if not property_obj or property_obj.owner_id != current_user.id:
                    raise HTTPException(
                        status_code=status.HTTP_403_FORBIDDEN,
                        detail="Access denied"
                    )
        
        return inspection
        
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error fetching inspection: {str(e)}"
        )

@router.put("/{inspection_id}", response_model=InspectionResponse)
async def update_inspection(
    inspection_id: int,
    inspection_update: InspectionUpdate,
    current_user: User = Depends(require_roles(["admin", "owner"])),
    db: Session = Depends(get_db)
):
    """Update an existing inspection"""
    try:
        inspection = next((i for i in inspections_db if i["id"] == inspection_id), None)
        if not inspection:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail="Inspection not found"
            )
        
        # Check access permissions
        if current_user.role == "owner":
            unit = db.query(Unit).filter(Unit.id == inspection["unit_id"]).first()
            if unit:
                property_obj = db.query(Property).filter(Property.id == unit.property_id).first()
                if not property_obj or property_obj.owner_id != current_user.id:
                    raise HTTPException(
                        status_code=status.HTTP_403_FORBIDDEN,
                        detail="Access denied"
                    )
        
        # Update inspection
        update_data = inspection_update.dict(exclude_unset=True)
        for key, value in update_data.items():
            if value is not None:
                inspection[key] = value
        
        inspection["updated_at"] = datetime.utcnow().isoformat()
        
        return inspection
        
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error updating inspection: {str(e)}"
        )

@router.delete("/{inspection_id}")
async def delete_inspection(
    inspection_id: int,
    current_user: User = Depends(require_roles(["admin", "owner"])),
    db: Session = Depends(get_db)
):
    """Delete an inspection"""
    try:
        inspection = next((i for i in inspections_db if i["id"] == inspection_id), None)
        if not inspection:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail="Inspection not found"
            )
        
        # Check access permissions
        if current_user.role == "owner":
            unit = db.query(Unit).filter(Unit.id == inspection["unit_id"]).first()
            if unit:
                property_obj = db.query(Property).filter(Property.id == unit.property_id).first()
                if not property_obj or property_obj.owner_id != current_user.id:
                    raise HTTPException(
                        status_code=status.HTTP_403_FORBIDDEN,
                        detail="Access denied"
                    )
        
        # Remove inspection
        inspections_db.remove(inspection)
        
        return {"message": "Inspection deleted successfully"}
        
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error deleting inspection: {str(e)}"
        )

@router.get("/summary/stats", response_model=dict)
async def get_inspection_stats(
    current_user: User = Depends(require_roles(["admin", "owner", "manager"])),
    db: Session = Depends(get_db)
):
    """Get inspection statistics"""
    try:
        # Filter inspections based on user role
        if current_user.role == "owner":
            owner_properties = db.query(Property).filter(Property.owner_id == current_user.id).all()
            property_ids = [p.id for p in owner_properties]
            filtered_inspections = [
                inspection for inspection in inspections_db
                if any(unit.property_id in property_ids for unit in db.query(Unit).filter(Unit.id == inspection["unit_id"]))
            ]
        else:
            filtered_inspections = inspections_db
        
        # Calculate statistics
        stats = {
            "total_inspections": len(filtered_inspections),
            "scheduled": len([i for i in filtered_inspections if i["status"] == "scheduled"]),
            "completed": len([i for i in filtered_inspections if i["status"] == "completed"]),
            "cancelled": len([i for i in filtered_inspections if i["status"] == "cancelled"]),
            "in_progress": len([i for i in filtered_inspections if i["status"] == "in_progress"]),
            "routine": len([i for i in filtered_inspections if i["inspection_type"] == "routine"]),
            "move_in": len([i for i in filtered_inspections if i["inspection_type"] == "move_in"]),
            "move_out": len([i for i in filtered_inspections if i["inspection_type"] == "move_out"]),
            "emergency": len([i for i in filtered_inspections if i["inspection_type"] == "emergency"])
        }
        
        return stats
        
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Error fetching inspection stats: {str(e)}"
        )

File: frontend/src/pages/Inspections/Inspections.js
import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  Box,
  Typography,
  Card,
  CardContent,
  Grid,
  Button,
  Chip,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  TextField,
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  Alert,
  Paper,
  Tabs,
  Tab,
  IconButton,
  Tooltip,
  Fab,
  Avatar,
} from '@mui/material';
import {
  Add,
  CalendarToday,
  Home,
  Apartment,
  CheckCircle,
  Schedule,
  Cancel,
  Edit,
  Delete,
  Visibility,
  Assignment,
  Warning,
} from '@mui/icons-material';
import { DataGrid } from '@mui/x-data-grid';
// Removed complex date picker dependencies

const StatCard = ({ title, value, icon, color, subtitle }) => (
  <Card sx={{ height: '100%' }}>
    <CardContent>
      <Box sx={{ display: 'flex', alignItems: 'center', mb: 2 }}>
        <Avatar sx={{ bgcolor: color, mr: 2 }}>
          {icon}
        </Avatar>
        <Box sx={{ flexGrow: 1 }}>
          <Typography variant="h4" component="div">
            {value}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {title}
          </Typography>
          {subtitle && (
            <Typography variant="caption" color="text.secondary">
              {subtitle}
            </Typography>
          )}
        </Box>
      </Box>
    </CardContent>
  </Card>
);

const Inspections = () => {
  const dispatch = useDispatch();
  const { user } = useSelector((state) => state.auth);
  const { units } = useSelector((state) => state.units);
  const { properties } = useSelector((state) => state.properties);
  
  const [activeTab, setActiveTab] = useState(0);
  const [inspections, setInspections] = useState([]);
  const [openDialog, setOpenDialog] = useState(false);
  const [editingInspection, setEditingInspection] = useState(null);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  
  const [formData, setFormData] = useState({
    unit_id: '',
    inspection_date: new Date(),
    inspection_type: 'routine',
    inspector_name: '',
    status: 'scheduled',
    notes: '',
    findings: '',
    recommendations: '',
    photos: []
  });

  // Mock data for inspections - in real app, this would come from API
  useEffect(() => {
    const mockInspections = [
      {
        id: 1,
        unit_id: 1,
        unit_number: '101',
        property_name: 'Sunset Apartments',
        property_address: '123 Sunset Boulevard, Los Angeles, CA',
        inspection_date: '2025-10-25',
        inspection_type: 'routine',
        inspector_name: 'John Smith',
        status: 'scheduled',
        notes: 'Quarterly routine inspection',
        findings: '',
        recommendations: '',
        monthly_rent: 2500,
        bedrooms: 2,
        bathrooms: 2,
        created_at: '2025-10-20T10:00:00Z'
      },
      {
        id: 2,
        unit_id: 2,
        unit_number: '102',
        property_name: 'Sunset Apartments',
        property_address: '123 Sunset Boulevard, Los Angeles, CA',
        inspection_date: '2025-10-28',
        inspection_type: 'move_out',
        inspector_name: 'Jane Doe',
        status: 'completed',
        notes: 'Move-out inspection for departing tenant',
        findings: 'Minor wall damage, carpet needs cleaning',
        recommendations: 'Repair wall damage, deep clean carpet',
        monthly_rent: 2200,
        bedrooms: 1,
        bathrooms: 1,
        created_at: '2025-10-18T14:30:00Z'
      },
      {
        id: 3,
        unit_id: 3,
        unit_number: '201',
        property_name: 'Premium Rental Complex',
        property_address: '456 Premium Street, Los Angeles, CA',
        inspection_date: '2025-11-01',
        inspection_type: 'move_in',
        inspector_name: 'Mike Johnson',
        status: 'scheduled',
        notes: 'Pre-move-in inspection for new tenant',
        findings: '',
        recommendations: '',
        monthly_rent: 3000,
        bedrooms: 3,
        bathrooms: 2,
        created_at: '2025-10-19T09:15:00Z'
      }
    ];
    setInspections(mockInspections);
  }, []);

  const handleTabChange = (event, newValue) => {
    setActiveTab(newValue);
  };

  const handleOpenDialog = (inspection = null) => {
    if (inspection) {
      setEditingInspection(inspection);
      setFormData({
        unit_id: inspection.unit_id,
        inspection_date: new Date(inspection.inspection_date),
        inspection_type: inspection.inspection_type,
        inspector_name: inspection.inspector_name,
        status: inspection.status,
        notes: inspection.notes,
        findings: inspection.findings,
        recommendations: inspection.recommendations,
        photos: inspection.photos || []
      });
    } else {
      setEditingInspection(null);
      setFormData({
        unit_id: '',
        inspection_date: new Date(),
        inspection_type: 'routine',
        inspector_name: '',
        status: 'scheduled',
        notes: '',
        findings: '',
        recommendations: '',
        photos: []
      });
    }
    setOpenDialog(true);
  };

  const handleCloseDialog = () => {
    setOpenDialog(false);
    setEditingInspection(null);
    setFormData({
      unit_id: '',
      inspection_date: new Date(),
      inspection_type: 'routine',
      inspector_name: '',
      status: 'scheduled',
      notes: '',
      findings: '',
      recommendations: '',
      photos: []
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setLoading(true);
    
    // Simulate API call
    setTimeout(() => {
      const newInspection = {
        id: editingInspection ? editingInspection.id : Date.now(),
        ...formData,
        inspection_date: formData.inspection_date.toISOString().split('T')[0],
        unit_number: units.find(u => u.id === parseInt(formData.unit_id))?.unit_number || 'N/A',
        property_name: properties.find(p => p.id === units.find(u => u.id === parseInt(formData.unit_id))?.property_id)?.name || 'N/A',
        property_address: properties.find(p => p.id === units.find(u => u.id === parseInt(formData.unit_id))?.property_id)?.address || 'N/A',
        monthly_rent: units.find(u => u.id === parseInt(formData.unit_id))?.monthly_rent || 0,
        bedrooms: units.find(u => u.id === parseInt(formData.unit_id))?.bedrooms || 0,
        bathrooms: units.find(u => u.id === parseInt(formData.unit_id))?.bathrooms || 0,
        created_at: new Date().toISOString()
      };

      if (editingInspection) {
        setInspections(prev => prev.map(i => i.id === editingInspection.id ? newInspection : i));
      } else {
        setInspections(prev => [...prev, newInspection]);
      }

      setLoading(false);
      handleCloseDialog();
    }, 1000);
  };

  const handleDelete = (inspectionId) => {
    setInspections(prev => prev.filter(i => i.id !== inspectionId));
  };

  const getStatusColor = (status) => {
    switch (status) {
      case 'scheduled': return 'primary';
      case 'completed': return 'success';
      case 'cancelled': return 'error';
      case 'in_progress': return 'warning';
      default: return 'default';
    }
  };

  const getStatusIcon = (status) => {
    switch (status) {
      case 'scheduled': return <Schedule />;
      case 'completed': return <CheckCircle />;
      case 'cancelled': return <Cancel />;
      case 'in_progress': return <Edit />;
      default: return <CalendarToday />;
    }
  };

  const getInspectionTypeColor = (type) => {
    switch (type) {
      case 'routine': return 'info';
      case 'move_in': return 'success';
      case 'move_out': return 'warning';
      case 'emergency': return 'error';
      default: return 'default';
    }
  };

  // Filter inspections based on active tab
  const filteredInspections = inspections.filter(inspection => {
    switch (activeTab) {
      case 0: return inspection.status === 'scheduled';
      case 1: return inspection.status === 'completed';
      case 2: return inspection.status === 'cancelled';
      case 3: return inspection.status === 'in_progress';
      default: return true;
    }
  });

  // Get available units for rent (not occupied) - ONLY these can be inspected
  const availableUnits = units.filter(unit => unit.status === 'available');
  
  // Show warning if no available units
  const hasAvailableUnits = availableUnits.length > 0;

  const columns = [
    { field: 'id', headerName: 'ID', width: 70 },
    { field: 'unit_number', headerName: 'Unit', width: 100 },
    { field: 'property_name', headerName: 'Property', width: 200 },
    { field: 'inspection_date', headerName: 'Date', width: 120 },
    { field: 'inspection_type', headerName: 'Type', width: 120 },
    { field: 'inspector_name', headerName: 'Inspector', width: 150 },
    { field: 'status', headerName: 'Status', width: 120 },
    { field: 'monthly_rent', headerName: 'Rent', width: 100 },
    {
      field: 'actions',
      headerName: 'Actions',
      width: 150,
      renderCell: (params) => (
        <Box>
          <Tooltip title="View Details">
            <IconButton size="small" onClick={() => handleOpenDialog(params.row)}>
              <Visibility />
            </IconButton>
          </Tooltip>
          <Tooltip title="Edit">
            <IconButton size="small" onClick={() => handleOpenDialog(params.row)}>
              <Edit />
            </IconButton>
          </Tooltip>
          <Tooltip title="Delete">
            <IconButton size="small" onClick={() => handleDelete(params.row.id)}>
              <Delete />
            </IconButton>
          </Tooltip>
        </Box>
      ),
    },
  ];

  return (
    <Box sx={{ p: 3 }}>
        <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', mb: 3 }}>
          <Typography variant="h4" component="h1" sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
            <CalendarToday />
            Unit Inspections
          </Typography>
          <Button
            variant="contained"
            startIcon={<Add />}
            onClick={() => handleOpenDialog()}
            disabled={!hasAvailableUnits}
            sx={{ borderRadius: 2 }}
          >
            Schedule Inspection
          </Button>
        </Box>

        {error && (
          <Alert severity="error" sx={{ mb: 2 }}>
            {typeof error === 'string' ? error : JSON.stringify(error)}
          </Alert>
        )}

        {!hasAvailableUnits && (
          <Alert severity="warning" sx={{ mb: 2 }}>
            <Typography variant="body1" sx={{ fontWeight: 'bold', mb: 1 }}>
              No Available Units for Inspection
            </Typography>
            <Typography variant="body2">
              Inspections can only be scheduled for units that are <strong>available for rent</strong> (not occupied). 
              Currently, all your units are occupied. You can schedule inspections when units become available.
            </Typography>
          </Alert>
        )}

        {/* Summary Cards */}
        <Grid container spacing={3} sx={{ mb: 3 }}>
          <Grid item xs={12} sm={6} md={3}>
            <Card>
              <CardContent>
                <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
                  <Box>
                    <Typography color="textSecondary" gutterBottom>
                      Scheduled Inspections
                    </Typography>
                    <Typography variant="h4">
                      {inspections.filter(i => i.status === 'scheduled').length}
                    </Typography>
                  </Box>
                  <Schedule color="primary" sx={{ fontSize: 40 }} />
                </Box>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={12} sm={6} md={3}>
            <Card>
              <CardContent>
                <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
                  <Box>
                    <Typography color="textSecondary" gutterBottom>
                      Completed Inspections
                    </Typography>
                    <Typography variant="h4">
                      {inspections.filter(i => i.status === 'completed').length}
                    </Typography>
                  </Box>
                  <CheckCircle color="success" sx={{ fontSize: 40 }} />
                </Box>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={12} sm={6} md={3}>
            <Card>
              <CardContent>
                <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
                  <Box>
                    <Typography color="textSecondary" gutterBottom>
                      Available Units
                    </Typography>
                    <Typography variant="h4">
                      {availableUnits.length}
                    </Typography>
                  </Box>
                  <Apartment color="info" sx={{ fontSize: 40 }} />
                </Box>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={12} sm={6} md={3}>
            <Card>
              <CardContent>
                <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
                  <Box>
                    <Typography color="textSecondary" gutterBottom>
                      Total Inspections
                    </Typography>
                    <Typography variant="h4">
                      {inspections.length}
                    </Typography>
                  </Box>
                  <CalendarToday color="secondary" sx={{ fontSize: 40 }} />
                </Box>
              </CardContent>
            </Card>
          </Grid>
        </Grid>

        {/* Inspection Status Tabs */}
        <Paper sx={{ mb: 3 }}>
          <Tabs value={activeTab} onChange={handleTabChange} variant="fullWidth">
            <Tab label="Scheduled" />
            <Tab label="Completed" />
            <Tab label="Cancelled" />
            <Tab label="In Progress" />
          </Tabs>
        </Paper>


        {/* Data Grid View */}
        <Box sx={{ mt: 3 }}>
          <Typography variant="h6" sx={{ mb: 2 }}>
            All Inspections
          </Typography>
          <DataGrid
            rows={inspections}
            columns={columns}
            pageSize={10}
            rowsPerPageOptions={[10, 25, 50]}
            disableSelectionOnClick
            sx={{ height: 400 }}
          />
        </Box>

        {/* Inspection Dialog */}
        <Dialog open={openDialog} onClose={handleCloseDialog} maxWidth="md" fullWidth>
          <DialogTitle>
            {editingInspection ? 'Edit Inspection' : 'Schedule New Inspection'}
          </DialogTitle>
          <form onSubmit={handleSubmit}>
            <DialogContent>
              <Grid container spacing={2}>
                <Grid item xs={12} sm={6}>
                  <FormControl fullWidth required>
                    <InputLabel>Unit (Available for Rent Only)</InputLabel>
                    <Select
                      value={formData.unit_id}
                      onChange={(e) => setFormData({ ...formData, unit_id: e.target.value })}
                      label="Unit (Available for Rent Only)"
                    >
                      {availableUnits.length === 0 ? (
                        <MenuItem disabled>
                          No available units for inspection
                        </MenuItem>
                      ) : (
                        availableUnits.map((unit) => (
                          <MenuItem key={unit.id} value={unit.id}>
                            Unit {unit.unit_number} - {properties.find(p => p.id === unit.property_id)?.name} (Available)
                          </MenuItem>
                        ))
                      )}
                    </Select>
                  </FormControl>
                </Grid>
                
                <Grid item xs={12} sm={6}>
                  <TextField
                    fullWidth
                    label="Inspection Date"
                    type="date"
                    value={formData.inspection_date.toISOString().split('T')[0]}
                    onChange={(e) => setFormData({ ...formData, inspection_date: new Date(e.target.value) })}
                    InputLabelProps={{
                      shrink: true,
                    }}
                    required
                  />
                </Grid>
                
                <Grid item xs={12} sm={6}>
                  <FormControl fullWidth required>
                    <InputLabel>Inspection Type</InputLabel>
                    <Select
                      value={formData.inspection_type}
                      onChange={(e) => setFormData({ ...formData, inspection_type: e.target.value })}
                      label="Inspection Type"
                    >
                      <MenuItem value="routine">Routine</MenuItem>
                      <MenuItem value="move_in">Move-In</MenuItem>
                      <MenuItem value="move_out">Move-Out</MenuItem>
                      <MenuItem value="emergency">Emergency</MenuItem>
                    </Select>
                  </FormControl>
                </Grid>
                
                <Grid item xs={12} sm={6}>
                  <TextField
                    fullWidth
                    label="Inspector Name"
                    value={formData.inspector_name}
                    onChange={(e) => setFormData({ ...formData, inspector_name: e.target.value })}
                    required
                  />
                </Grid>
                
                <Grid item xs={12} sm={6}>
                  <FormControl fullWidth required>
                    <InputLabel>Status</InputLabel>
                    <Select
                      value={formData.status}
                      onChange={(e) => setFormData({ ...formData, status: e.target.value })}
                      label="Status"
                    >
                      <MenuItem value="scheduled">Scheduled</MenuItem>
                      <MenuItem value="in_progress">In Progress</MenuItem>
                      <MenuItem value="completed">Completed</MenuItem>
                      <MenuItem value="cancelled">Cancelled</MenuItem>
                    </Select>
                  </FormControl>
                </Grid>
                
                <Grid item xs={12}>
                  <TextField
                    fullWidth
                    label="Notes"
                    value={formData.notes}
                    onChange={(e) => setFormData({ ...formData, notes: e.target.value })}
                    multiline
                    rows={3}
                  />
                </Grid>
                
                <Grid item xs={12}>
                  <TextField
                    fullWidth
                    label="Findings"
                    value={formData.findings}
                    onChange={(e) => setFormData({ ...formData, findings: e.target.value })}
                    multiline
                    rows={3}
                    placeholder="Enter inspection findings..."
                  />
                </Grid>
                
                <Grid item xs={12}>
                  <TextField
                    fullWidth
                    label="Recommendations"
                    value={formData.recommendations}
                    onChange={(e) => setFormData({ ...formData, recommendations: e.target.value })}
                    multiline
                    rows={3}
                    placeholder="Enter recommendations..."
                  />
                </Grid>
              </Grid>
            </DialogContent>
            <DialogActions>
              <Button onClick={handleCloseDialog}>Cancel</Button>
              <Button type="submit" variant="contained" disabled={loading}>
                {loading ? 'Saving...' : editingInspection ? 'Update' : 'Schedule'}
              </Button>
            </DialogActions>
          </form>
        </Dialog>
      </Box>
    );
  };

  export default Inspections;